### PR TITLE
feat: worked on getting customer specific information endpoints

### PR DESCRIPTION
**Pull request by samumaretiya**

### CODE REVIEW

File: Model/CustomerInformation.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+            // Retrieve additional customer information
+            $customersCollection = $this->customerCollection->getCollection()
+                ->addAttributeToSelect("*")
+                ->addAttributeToFilter("entity_id", $customerId)
+                ->load();
+            $customerInformationAll = $customersCollection->getData();
+            $customerInformation = array();
+            foreach ($customerInformationAll as $customerInfo) {
+                unset($customerInfo['password_hash']);
```
Comment (soleilcot) [inline Model/CustomerInformation.php:80]: Let's also remove `rp_token`, we don't need it so it's best not to expose it.

File: Model/CustomerInformation.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        try {
+            $body       = $this->request->getBodyParams();
+            $customerId = $body['data']['customer_id'];
+
+            // Retrieve customer information by ID
+            $customer      = $this->customerRepository->getById($customerId);
+            $id            = $customer->getId();
+            $email         = $customer->getEmail();
+            $createdAt     = $customer->getCreatedAt();
+            $createdDate   = date('m/d/Y', strtotime($createdAt));
```
Comment (soleilcot) [inline Model/CustomerInformation.php:49]: SonarCloud suggests using a class constant for m/d/Y

File: Model/CustomerInformation.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            $lastLoginAt   = $this->customerLog->get($customerId);
+            $lastLoginDate = date('m/d/Y', strtotime($lastLoginAt->getLastLoginAt()));
```
Comment (soleilcot) [inline Model/CustomerInformation.php:51]: SonarCloud suggests using a class constant for m/d/Y

File: Model/CustomerInformation.php [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            $remoteIp               = "";
+            $lastOrderDateFormatted = "";
+            $totalOrders            = "";
+            $totalOrderAmount       = "";
+            if (count($orderCollection)) {
+                // Retrieve details from the last order
+                $lastOrder     = $orderCollection->getFirstItem();
+                $remoteIp      = $lastOrder->getRemoteIp();
+                $lastOrderDate = $lastOrder->getCreatedAt();
+                $lastOrderDateFormatted = date('m/d/Y', strtotime($lastOrderDate));
```
Comment (soleilcot) [inline Model/CustomerInformation.php:66]: SonarCloud suggests using a class constant for m/d/Y

File: Model/CustomerInformation.php [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+            // Retrieve customer's saved credit card date
+            $cardList = $this->paymenttokenmanagement->getListByCustomerId($customerId);
+            $customerLastUsedCardDate = "";
+            if ($cardList) {
+                // Retrieve the last used card's date
+                foreach ($cardList as $card) {
+                    $customerUsedCards = $card->getData();
+                    end($customerUsedCards);
+                    $customerLastUsedCardDate = date('m/d/Y', strtotime($customerUsedCards['created_at']));
```
Comment (soleilcot) [inline Model/CustomerInformation.php:92]: SonarCloud suggests using a class constant for m/d/Y

**Review by soleilcot — CHANGES_REQUESTED** [review]

File: Model/CustomerInformation.php (hunk 1 above)
Comment (samumaretiya) [reply Model/CustomerInformation.php:87]: Hi @soleilcot , I've added the recommenedation. see here https://github.com/Nofraud/nofraud_magento_checkout/pull/41/files

File: Model/CustomerInformation.php (hunk 2 above)
Comment (samumaretiya) [reply Model/CustomerInformation.php]: Hi @soleilcot , I've added the recommenedation. see here https://github.com/Nofraud/nofraud_magento_checkout/pull/41/files

File: Model/CustomerInformation.php (hunk 3 above)
Comment (samumaretiya) [reply Model/CustomerInformation.php]: Hi @soleilcot , I've added the recommenedation. see here https://github.com/Nofraud/nofraud_magento_checkout/pull/41/files

File: Model/CustomerInformation.php (hunk 4 above)
Comment (samumaretiya) [reply Model/CustomerInformation.php]: Hi @soleilcot , I've added the recommenedation. see here https://github.com/Nofraud/nofraud_magento_checkout/pull/41/files

File: Model/CustomerInformation.php (hunk 5 above)
Comment (samumaretiya) [reply Model/CustomerInformation.php]: Hi @soleilcot , I've added the recommenedation. see here https://github.com/Nofraud/nofraud_magento_checkout/pull/41/files

**Pull request merged**